akka-http-core/src/main/resources/reference.conf

```diff
@@ -278,6 +278,18 @@ akka.http {
 
       # Enable verbose debug logging for all ingoing and outgoing frames
       log-frames = false
+
+      # When there is no data transmitted, send a HTTP/2 ping frame with this interval to make sure the connection is
+      # kept alive, 0s disables sending ping frames. Must be full seconds.
+      keepalive-time = 0s
+
+      # Fail the connection if a ping frame is sent and does not come back from the server within this timeout.
+      # Must be full seconds and less than the ping interval.
+      keepalive-timeout = 5s
+
+      # Fail the connection if more ping interchanges than this are done without any data or header frames.
+      # Setting to 0 disables this limit. Only in effect if ping-interval is enabled.
+      max-keepalives-without-data = 300000
```

Comment:
Where does that default value come from?

Reply:
those names/defaults were inspired by https://github.com/grpc/grpc/blob/master/doc/keepalive.md

Reply:
For naming I started out with ping as that is the protocol feature but then after some feedback from raboof I reconsidered and thought that it was better to name it after the purpose, hoping that is what I'd be looking for if I wanted to use it. It's not a super convinced stance though.

Reply:
I see. Keep-alive is just one of the purposes, though, the other one is detecting when a connection is gone. If we want to name it after that, those names seem a bit misleading. `keepalive-time` is not the time the connection is kept alive and `keepalive-timeout` is misleading because it's not the timeout after which the keepalive is stopped.